examples/scripts/spme_max_energy.py

```diff
@@ -0,0 +1,175 @@
+    ),
+]
+
+
+# Define functions
+def nominal_capacity(x, model):
+    """
+    Update the nominal capacity based on the theoretical energy density and an
+    average voltage.
+    """
+    inputs = {
+        key: x[i] for i, key in enumerate([param.name for param in model.parameters])
+    }
+    model._parameter_set.update(inputs)
+
+    theoretical_energy = model._electrode_soh.calculate_theoretical_energy(
+        model._parameter_set
+    )
+    average_voltage = (
+        model._parameter_set["Upper voltage cut-off [V]"]
+        + model._parameter_set["Lower voltage cut-off [V]"]
+    ) / 2
+    theoretical_capacity = theoretical_energy / average_voltage
+    model._parameter_set.update({"Nominal cell capacity [A.h]": theoretical_capacity})
+
+
+def cell_mass(ps):
+    """
+    Compute the total cell mass [kg] for the current parameter set.
+    """
+
+    # Approximations due to SPM(e) parameter set limitations
+    electrolyte_density = ps["Separator density [kg.m-3]"]
+
+    # Electrode mass densities [kg.m-3]
+    positive_mass_density = (
+        ps["Positive electrode active material volume fraction"]
+        * ps["Positive electrode density [kg.m-3]"]
+    )
+    +(ps["Positive electrode porosity"] * electrolyte_density)
+    negative_mass_density = (
+        ps["Negative electrode active material volume fraction"]
+        * ps["Negative electrode density [kg.m-3]"]
+    )
+    +(ps["Negative electrode porosity"] * electrolyte_density)
+
+    # Areal mass densities [kg.m-2]
+    positive_mass = ps["Positive electrode thickness [m]"] * positive_mass_density
+    negative_mass = ps["Negative electrode thickness [m]"] * negative_mass_density
+    separator_mass = (
+        ps["Separator thickness [m]"]
+        * ps["Separator porosity"]
+        * ps["Separator density [kg.m-3]"]
+    )
+    positive_current_collector_mass = (
+        ps["Positive current collector thickness [m]"]
+        * ps["Positive current collector density [kg.m-3]"]
+    )
+    negative_current_collector_mass = (
+        ps["Negative current collector thickness [m]"]
+        * ps["Negative current collector density [kg.m-3]"]
+    )
+
+    # Cross-sectional area [m2]
+    cross_sectional_area = ps["Electrode height [m]"] * ps["Electrode width [m]"]
+
+    return cross_sectional_area * (
+        positive_mass
+        + separator_mass
+        + negative_mass
+        + positive_current_collector_mass
+        + negative_current_collector_mass
+    )
+
+
+# Define test protocol
+experiment = pybamm.Experiment(
+    ["Discharge at 1C until 2.5 V (5 seconds period)"],
+)
+init_soc = 1  # start from full charge
+signal = ["Voltage [V]", "Current [A]"]
+
+# Generate problem
+problem = pybop.DesignProblem(
+    model, parameters, experiment, signal=signal, init_soc=init_soc
+)
+
+# Update the C-rate and the example dataset
+nominal_capacity(problem.x0, model)
+sol = problem.evaluate(problem.x0)
+problem._time_data = sol[:, -1]
```

Comment:
Something else! For the signal I'm using current and voltage, but we need time as well for plotting, so in DesignProblem's evaluate function, I add time to the list of signals and the time_data as an extra column in the output array so that it can be passed back and stored. So here `-1` means last column.